mettle/worker/worker.go

```diff
@@ -550,6 +550,14 @@ func (w *worker) execute(action *pb.Action, command *pb.Command) *pb.ExecuteResp
 	uploadDurations.Observe(end.Sub(execEnd).Seconds())
 	log.Notice("Uploaded outputs for %s", w.actionDigest.Hash)
 	w.metadata.WorkerCompletedTimestamp = toTimestamp(time.Now())
+
+	if req.SkipCacheLookup {
+		return &pb.ExecuteResponse{
```

Comment:
so I don't think this is quite right, we should still try to update it in case one doesn't exist, but if `SkipCacheLookup` we should return our one rather than what the server sent back.

Having said that, that is still not quite right, because we should be able to tell the server to overwrite the action which we currently can't / don't.

Reply:
Okay, I've updated that. I think the correct way to go about this is as I described in the PR description. Let me know hat you think.

Reply:
Returning the existing result is deliberate to improve determinism (and I think is implicit in the API by the fact that the RPC returns an ActionResult, which suggests it doesn't have to be the one you sent over).

Think this looks good as is, but agreed we should figure out a way to allow overwriting those results in at least some cases.